plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -808,6 +811,12 @@ private Optional<List<String>> doGetPartitionNames(
             case PRE_EXISTING_TABLE -> getOptionalPartitions(databaseName, tableName, columnNames, partitionKeysFilter)
                     .orElseThrow(() -> new TrinoException(TRANSACTION_CONFLICT, format("Table '%s.%s' was dropped by another transaction", databaseName, tableName)));
         };
+        Set<String> duplicatePartitionNames = partitionNames.stream()
```

Comment:
This would be slightly simpler using a `Multiset`:
```java
Set<String> duplicatePartitionNames = ImmutableMultiset.copyOf(partitionNames)
        .entrySet().stream()
        .filter(entry -> entry.getCount() > 1)
        .map(Multiset.Entry::getElement)
        .collect(toImmutableSet());